src/Microsoft.Diagnostics.NETCore.Client/ReversedServer/ReversedDiagnosticsServer.cs

```diff
@@ -46,33 +53,25 @@ internal sealed class ReversedDiagnosticsServer : IDisposable
         /// On all other systems, this must be the full file path of the socket.
         /// </param>
         public ReversedDiagnosticsServer(string transportPath)
-            : this(transportPath, MaxAllowedConnections)
         {
+            _transportPath = transportPath;
         }
 
-        /// <summary>
-        /// Constructs the <see cref="ReversedDiagnosticsServer"/> instance with an endpoint bound
-        /// to the location specified by <paramref name="transportPath"/>.
-        /// </summary>
-        /// <param name="transportPath">
-        /// The path of the server endpoint.
-        /// On Windows, this can be a full pipe path or the name without the "\\.\pipe\" prefix.
-        /// On all other systems, this must be the full file path of the socket.
-        /// </param>
-        /// <param name="maxConnections">The maximum number of connections the server will support.</param>
-        public ReversedDiagnosticsServer(string transportPath, int maxConnections)
-        {
-            _transport = IpcServerTransport.Create(transportPath, maxConnections);
-        }
-
-        public void Dispose()
+        public async ValueTask DisposeAsync()
         {
             if (!_disposed)
             {
                 _disposalSource.Cancel();
 
-                lock (_lock)
+                if (null != _listenTask)
+                {
+                    await _listenTask.ConfigureAwait(false);
```

Comment:
Do we need to catch here, in case the task fails or throws out an exception?

Reply:
Changed to catch and Debug.Fail with message.